### PR TITLE
Added 'state' parameter to step1_get_authorize_url.

### PR DESCRIPTION
**Pull request by frederikcreemers**
This PR adds a state parameter to step1_get_authorize_url. It is added as a query parameter to the autorization URL as described in the OAuth 2.0 spec.

See issue #170

### CODE REVIEW

File: tests/test_oauth2client.py [hunk 1]
```diff
@@ -989,7 +989,7 @@ def test_construct_authorize_url(self):
     self.assertEqual('code', q['response_type'][0])
     self.assertEqual('foo', q['scope'][0])
     self.assertEqual(OOB_CALLBACK_URN, q['redirect_uri'][0])
-    self.assertEqual('offline', q['access_type'][0])
```
Comment (dhermes) [inline tests/test_oauth2client.py:992]: Why remove this check? You can still check both the `access_type` and the `state`.

File: .gitignore [hunk 2]
```diff
@@ -21,3 +21,6 @@ scripts/local_test_setup
 tests/data/key.json
 tests/data/key.p12
 tests/data/user-key.json
+
+# PyCharm configuration:
+.idea
```
Comment (anthmgoogle) [inline .gitignore:26]: This seems unrelated. Perhaps factor into a separate pull request?

File: tests/test_oauth2client.py (hunk 1 above)
Comment (frederikcreemers) [inline tests/test_oauth2client.py:992]: @dhermes Sorry, that was accidental. Fixing that...

**Comment by dhermes** [comment]
@bigblind I'm going to merge #221 once it passes Travis. Can you rebase this PR on top of it at that time?

**Comment by dhermes** [comment]
@bigblind #221 is in, you can rebase now.

**Comment by bigblind** [comment]
@dhermes rebased.

**Comment by dhermes** [comment]
This all (all 5 lines!) LGTM. I'll wait on Travis to pass and @anthmgoogle to comment before we merge it.

**Comment by anthmgoogle** [comment]
LGTM.

Thanks for this.
